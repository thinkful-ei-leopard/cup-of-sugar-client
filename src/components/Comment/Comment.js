import React from 'react';
import Button from '../Button/Button';
import UserContext from '../../contexts/UserContext';
import styles from './Comment.module.scss';
import CommentsApiService from '../../services/comments-api-service';
import Confirm from '../Confirm/Confirm';
import '@reach/dialog/styles.css';

export default class Comment extends React.Component {
  static contextType = UserContext;

  render() {
    const user = this.context.user;
    const comment = this.props.comment;

    let deleteButton =
      user.id === comment.user_id ? (
        <Confirm title="Confirm" description="Are you sure?">
          {(confirm) => (
            <Button
              onClick={confirm(() => {
                CommentsApiService.handleCommentDelete(
                  this.props.comment.id,
                  this.props.comment.post_id
                );
                this.props.deleteComment(this.props.comment.id);
              })}
              type="delete"
              title="Delete"
              className={styles.deletePostButton}
              id={styles.deleteCommentButton}>
              delete
            </Button>
          )}
        </Confirm>
      ) : null;

    return (
      <li className={styles.commentLi}>
        <div className={styles.commentDeleteContainer}>
          <p className={styles.content}>
            {comment.content} <br />
            {' '}
            <span className={styles.commenterName}>
              {comment.name} ({comment.user_name})
            </span>{' '}
            <span className={styles.datePosted}>
              {comment.date_modified.slice(0, 10)}
            </span>
          </p>
          {deleteButton}
        </div>
      </li>
    );
  }
}
