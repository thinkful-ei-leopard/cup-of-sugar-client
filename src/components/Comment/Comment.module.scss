@import '../../styles/_variables';
@import '../../styles/_breakpoints';

.name {
  display: inline-flex;
  margin-left: 0%;
  margin-right: 10%;
  margin-bottom: -1em;
}

#deleteCommentButton {
  padding: 0.25rem 0.5rem;
  color: var(--black);
  margin-left: 0.1rem;
  background-color: transparent;
  border: none;
  font-weight: bold;
  font-size: 0.7rem;
  transition: color 0.2s;
  &:hover {
    color: red;
  }
}

.commenterName {
  margin: 0 auto;
  text-align: center;
  font-size: 1rem;
  font-style: italic;
  color: var(--dark2);
}

.commentDeleteContainer {
  display: flex;
  flex-direction: column;
}

.commentLi {
  margin-bottom: 1rem;
}

.datePosted {
    font-size: .9rem;
    font-style: italic;
    opacity: .5;
}


