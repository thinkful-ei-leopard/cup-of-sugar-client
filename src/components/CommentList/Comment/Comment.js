import React from 'react';
import Button from '../../Button/Button';
import UserContext from '../../../contexts/UserContext';
import styles from './Comment.module.scss';
import CommentsApiService from '../../../services/comments-api-service';
import Confirm from '../../Confirm/Confirm';
import '@reach/dialog/styles.css';
import { withRouter } from 'react-router-dom';
import MessageUser from '../../../components/MessageUser/MessageUser';

class Comment extends React.Component {
  state = { loading: true };

  static contextType = UserContext;

  componentDidMount() {
    this.setState({ loading: false });
  }

  render() {
    const { user } = this.context;
    const { comment } = this.props;

    if (this.state.loading == true) {
      return <></>;
    }

    let deleteButton =
      user.id === comment.user_id ? (
        <Confirm title="Delete" description="Are you sure?">
          {confirm => (
            <Button
              onClick={confirm(() => {
                CommentsApiService.handleCommentDelete(
                  this.props.comment.id,
                  this.props.comment.post_id
                );
                this.props.deleteComment(this.props.comment.id);
              })}
              type="delete"
              title="Delete"
              className={styles.deletePostButton}
              id={styles.deleteCommentButton}>
              delete
            </Button>
          )}
        </Confirm>
      ) : null;

    return (
      <li className={styles.commentLi}>
        <div className={styles.commentDeleteContainer}>
          <p className={styles.content}>
            {comment.content} <br />{' '}
            <span className={styles.commenterName}>
              {comment.name} ({comment.user_name})
            </span>{' '}
            <span className={styles.datePosted}>
              {comment.date_modified.slice(0, 10)}
            </span>
          </p>
          {deleteButton}
          <MessageUser
            user={this.context.user}
            neighborId={this.props.comment.user_id}
            text={'message'}
          />
        </div>
      </li>
    );
  }
}

export default withRouter(Comment);
