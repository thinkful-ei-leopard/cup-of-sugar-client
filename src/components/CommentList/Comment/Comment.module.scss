@import '../../../styles/_variables';
@import '../../../styles/_breakpoints';

.name {
  display: inline-flex;
  margin-left: 0%;
  margin-right: 10%;
  margin-bottom: -1em;
}

.messageNeighborButton {
  padding: 0.25rem 0.5rem;
  color: var(--black);
  margin-left: 0.1rem;
  background-color: transparent;
  border: none;
  font-weight: bold;
  font-size: 0.7rem;
  transition: color 0.2s;
  &:hover, &focus {
    color: var(--dark1);
    cursor: pointer;
  }
}

#deleteCommentButton {
  padding: 0.25rem 0.5rem;
  color: var(--black);
  margin-left: 0.1rem;
  background-color: transparent;
  border: none;
  font-weight: bold;
  font-size: 0.7rem;
  transition: color 0.2s;
  &:hover, &focus {
    color: var(--warning);
    cursor: pointer;
  }
}

.commenterName {
  margin: 0 auto;
  text-align: center;
  font-size: 1rem;
  font-style: italic;
  color: var(--dark2);
}

.commentDeleteContainer {
  display: flex;
  flex-direction: column;
}

.commentLi {
  margin-bottom: 1.5rem;
}

.datePosted {
  font-size: 0.9rem;
  font-style: italic;
  opacity: 0.5;
}

.content {
  word-wrap: break-word;
}
