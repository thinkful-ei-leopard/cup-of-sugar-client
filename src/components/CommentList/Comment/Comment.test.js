import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Comment from './Comment';

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Router>
      <Comment
        comment={{ date_modified: '2020-05-04T19:25:55.864Z' }}
        deleteComment={() => {}}
      />
    </Router>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Comment
          comment={{ date_modified: '2020-05-04T19:25:55.864Z' }}
          deleteComment={() => {}}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
