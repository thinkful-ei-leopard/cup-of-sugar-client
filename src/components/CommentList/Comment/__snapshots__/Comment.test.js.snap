// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<li
  className="commentLi"
>
  <div
    className="commentDeleteContainer"
  >
    <p
      className="content"
    >
       
      <br />
       
      <span
        className="commenterName"
      >
         (
        )
      </span>
       
      <span
        className="datePosted"
      >
        2020-05-04
      </span>
    </p>
    <button
      className="Button deletePostButton"
      id="deleteCommentButton"
      onClick={[Function]}
      title="Delete"
      type="delete"
    >
      delete
    </button>
  </div>
</li>
`;
